# Problema 1

Las matrices $A$ y $B$ representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, $a_{ij}$ y $b_{ij}$ representan el monto de las ventas del producto $i$ en la región $j$ el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz $S = A + B$.
3. Calcule la matriz $D = B - A$.
4. Indique el valor e interprete los elementos $s_{24}$ y $d_{24}$.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz $E$. ¿Es posible determinar la matriz $T = A + B + E$? Justifique su respuesta.

Desarrollo
1.Orden de la matriz 
3X4

In [ ]:
#pregunta 2
#primero en cada variable asignamos una matriz, luego creamos otra varaible "s" y aquí asignamos la suma.
import numpy as np
A = np.array([[2.6, 4.8, 1.8, 0.9],
[3.2, 4.4, 2.5, 2.8],
[2.4, 3.6, 3.8, 2.5]])

B = np.array([[3.6, 2.5, 3, 2.5],
[4.5, 5, 3.5, 3.8],
[2.9, 3.0, 4.6, 4]])

S = A + B
print(S)


[[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]]


In [2]:
#pregunta 3

import numpy as np

D= B - A

print(D)

[[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]]


In [9]:
#pregunta 4

print (f'El valor en la posicion es {S[1][3]}')
print (f'La interpretacion es: El total de ventas del producto 2 en la region 4 es de: {S[1][3]} millones de dolares')

print (f'La interpretacion es: El segundo año se vendio {D[1][3]} millones de dolares mas del producto en la region 4')

El valor en la posicion es 6.6
La interpretacion es: El total de ventas del producto 2 en la region 4 es de: 6.6 millones de dolares
La interpretacion es: El segundo año se vendio 1.0 millones de dolares mas del producto en la region 4


pregunta 5

son matrizes de distinto orden y no se pueden sumar 

# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

pregunta 1

c1 c2 c3 = componentes
m1 m2 m3 m4 m5 = modelos de pc

Numeros = miles de dolares


In [ ]:
#pregunta 1

import numpy as np

J = np.array([[100, 200, 150],
[120, 210, 170],
[112, 200, 131],
[124, 310, 170],
[136, 156, 140]])

C = np.array([[110, 190, 160],
[130, 220, 180],
[146, 190, 141],
[143, 312, 163],
[137, 116, 156]])

#Orden de la Matriz

print(f'El orden de la matriz J es {J.shape}')
print(f'El orden de la matriz C es {C.shape}')

El orden de la matriz J es (5, 3)
El orden de la matriz C es (5, 3)


In [8]:
#pregunta 2

#Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.

print(f'El elemento j_21 es {J[1][0]}')
print(f'El elemento C_12 es {J[0][1]}\n')

print(f'Interpretacion: Los costos asociados al modelo 2, componente 1 en japon es de {J[1][0]} miles de dolares')
print(f'Interpretacion: Los costos asociados al modelo 1, componente 2 en china es de {J[0][1]} miles de dolares')

El elemento j_21 es 120
El elemento C_12 es 200

Interpretacion: Los costos asociados al modelo 2, componente 1 en japon es de 120 miles de dolares
Interpretacion: Los costos asociados al modelo 1, componente 2 en china es de 200 miles de dolares


In [10]:
#pregunta 3

# 3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.

S = J + C
print(f'El costo total entre ambas sucursales es: \n {S} \n')

El costo total entre ambas sucursales es: 
 [[210 390 310]
 [250 430 350]
 [258 390 272]
 [267 622 333]
 [273 272 296]] 



pregunta 4

J43
C43

In [14]:
#pregunta 4

D = J - C
print(f'La diferencia pedida es {D[3][2]} miles de dolares, por lo que el trabajador tiene razon')

La diferencia pedida es 7 miles de dolares, por lo que el trabajador tiene razon


# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

In [6]:
import numpy as np
C = np.array([[120, 200, 155, 125,230],
[155, 130, 110, 160, 180],
[140, 205, 150, 125, 155],
[130, 120, 230, 225, 125],
[135, 70, 190, 175, 95],
[110, 265, 180, 230, 180],
[185, 85, 120, 135, 150]])

#pregunta 1
E = 0.8*C
print(f'\n 1. La matriz correspondiente a Ecuador es: \n{E}')

#pregunta 2
# La expresion matricial

print(f'\n 2. La expresion correspondiente es t_ij = c_ij + e_ij')

#3 sumar

T = C + E
print(f'\n 3. La matriz correspondiente a la suma es \n{T}')

#4

print(f'\n 4. La cantidad de unidades del producto 3 fabricadas en la planta 4 en Chile es {C[2][3]} unidades')
T[6][1]


 1. La matriz correspondiente a Ecuador es: 
[[ 96. 160. 124. 100. 184.]
 [124. 104.  88. 128. 144.]
 [112. 164. 120. 100. 124.]
 [104.  96. 184. 180. 100.]
 [108.  56. 152. 140.  76.]
 [ 88. 212. 144. 184. 144.]
 [148.  68.  96. 108. 120.]]

 2. La expresion correspondiente es t_ij = c_ij + e_ij

 3. La matriz correspondiente a la suma es 
[[216. 360. 279. 225. 414.]
 [279. 234. 198. 288. 324.]
 [252. 369. 270. 225. 279.]
 [234. 216. 414. 405. 225.]
 [243. 126. 342. 315. 171.]
 [198. 477. 324. 414. 324.]
 [333. 153. 216. 243. 270.]]

 4. La cantidad de unidades del producto 3 fabricadas en la planta 4 en Chile es 125 unidades


np.float64(153.0)

# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

In [33]:
import numpy as np

V = np.array([[30, 34, 20],
[45, 20, 16],
[14, 26, 25]])

T = np.array([[35, 30, 26],
[52, 25, 18],
[23, 24, 32]])

#pregunta 1 sumar
A = V + T
print(f'\n 1. La produccion mensual para ambas plantas corresponde al \n{A}')

#pregunta 2: Interprete el elemento $a_{21}$ en el contexto del problema.
print(f'\n 2. La produccion de zapatos blancos para niños es de {A[1][0]} miles de pares\n')

#3

B = 1.3*V + 0.9*T
print(f'\n 3. La nueva matriz es \n {B}\n')

#4

print(f'\n 4. Para el proximo año se estima la produccion de zapatos negros para caballeros es de {B[0][2]:.1f} miles de pares\n')


 1. La produccion mensual para ambas plantas corresponde al 
[[65 64 46]
 [97 45 34]
 [37 50 57]]

 2. La produccion de zapatos blancos para niños es de 97 miles de pares


 3. La nueva matriz es 
 [[ 70.5  71.2  49.4]
 [105.3  48.5  37. ]
 [ 38.9  55.4  61.3]]


 4. Para el proximo año se estima la produccion de zapatos negros para caballeros es de 49.4 miles de pares



# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

In [37]:
import numpy as np

M = np.fromfunction(lambda i, j: 8*(i+1) + 12*(j+1), (8,9))
print(f'1. La matriz buscada es \n{M}\n')

#matriz transpuesta
print(f'2. La matriz transpuesta buscada es: \n{M.T}\n')

1. La matriz buscada es 
[[ 20.  32.  44.  56.  68.  80.  92. 104. 116.]
 [ 28.  40.  52.  64.  76.  88. 100. 112. 124.]
 [ 36.  48.  60.  72.  84.  96. 108. 120. 132.]
 [ 44.  56.  68.  80.  92. 104. 116. 128. 140.]
 [ 52.  64.  76.  88. 100. 112. 124. 136. 148.]
 [ 60.  72.  84.  96. 108. 120. 132. 144. 156.]
 [ 68.  80.  92. 104. 116. 128. 140. 152. 164.]
 [ 76.  88. 100. 112. 124. 136. 148. 160. 172.]]

2. La matriz transpuesta buscada es: 
[[ 20.  28.  36.  44.  52.  60.  68.  76.]
 [ 32.  40.  48.  56.  64.  72.  80.  88.]
 [ 44.  52.  60.  68.  76.  84.  92. 100.]
 [ 56.  64.  72.  80.  88.  96. 104. 112.]
 [ 68.  76.  84.  92. 100. 108. 116. 124.]
 [ 80.  88.  96. 104. 112. 120. 128. 136.]
 [ 92. 100. 108. 116. 124. 132. 140. 148.]
 [104. 112. 120. 128. 136. 144. 152. 160.]
 [116. 124. 132. 140. 148. 156. 164. 172.]]



# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



In [46]:
import numpy as np

A= np.fromfunction(lambda i, j: (i+1) * (j+1), (5,7))
print(f'La matriz A es: \n{A}\n')

B= np.fromfunction(lambda i, j: 15*(i+1)  - 2*(j+1), (7,5))
print(f'La matriz B es: \n{B}\n')

C = A.T + 0.5*B
print(f'La transpuesta de C es: \n{C}\n')

La matriz A es: 
[[ 1.  2.  3.  4.  5.  6.  7.]
 [ 2.  4.  6.  8. 10. 12. 14.]
 [ 3.  6.  9. 12. 15. 18. 21.]
 [ 4.  8. 12. 16. 20. 24. 28.]
 [ 5. 10. 15. 20. 25. 30. 35.]]

La matriz B es: 
[[ 13.  11.   9.   7.   5.]
 [ 28.  26.  24.  22.  20.]
 [ 43.  41.  39.  37.  35.]
 [ 58.  56.  54.  52.  50.]
 [ 73.  71.  69.  67.  65.]
 [ 88.  86.  84.  82.  80.]
 [103. 101.  99.  97.  95.]]

La transpuesta de C es: 
[[ 7.5  7.5  7.5  7.5  7.5]
 [16.  17.  18.  19.  20. ]
 [24.5 26.5 28.5 30.5 32.5]
 [33.  36.  39.  42.  45. ]
 [41.5 45.5 49.5 53.5 57.5]
 [50.  55.  60.  65.  70. ]
 [58.5 64.5 70.5 76.5 82.5]]



# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


°C, %, hp 
Lu, Ma, Mi, Ju, Vi

In [50]:
import numpy as np

D = np.array([[20, 28, 22, 26, 24],
[52, 57, 56, 54, 58],
[1066, 1068, 1066, 1064, 1068]])

#1

S = 0.95*D
print(f'La nueva matriz es: \n{S}\n')

#2 Matriz transpuerta
print(f'La nueva matriz es \n{S.T}\n')

#3
print(f'El porcentaje de humedad registrado el dia jueves es de {S[1][3]}%')
print(f'La Temperatura  registrada el dia miercoles es de {S.T[2][0]}°C')

La nueva matriz es: 
[[  19.     26.6    20.9    24.7    22.8 ]
 [  49.4    54.15   53.2    51.3    55.1 ]
 [1012.7  1014.6  1012.7  1010.8  1014.6 ]]

La nueva matriz es 
[[  19.     49.4  1012.7 ]
 [  26.6    54.15 1014.6 ]
 [  20.9    53.2  1012.7 ]
 [  24.7    51.3  1010.8 ]
 [  22.8    55.1  1014.6 ]]

El porcentaje de humedad registrado el dia jueves es de 51.3%
La Temperatura  registrada el dia miercoles es de 20.9°C


# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.
